### ISSUE TITLE
ReplicationController failed to notice Pod Eviction

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#15835]
TL;DR: I don't have a reliable repro yet, it might be a one-off flake. If it's not it can be serious, as we can permanently kill a Pod.

I was running 250 Node tests (denisty 30') looking at heapster resource consumption. The first run was close to the limit (3G), so the second one was likely to end in OOM. To nobody's surprise it did. Because it was eating by far most memory it was (most likely) killed by the sys oom killer, which, for some reason ended up in:
```
Mon, 19 Oct 2015 11:11:11 +0200   Mon, 19 Oct 2015 11:11:11 +0200   1         heapster-v10-j8a4y                                  Pod                                                NodeControllerEviction   {controllermanager }                    Marking for deletion Pod heapster-v10-j8a4y from Node e2e-test-gmarek-minion-9ei
```
(Node itself was healthy at the moment)

Small problem is that the Pod is not running anywhere now:
```
kubectl --namespace=kube-system get pod | grep heapster
```
returns an empty result.

The real question is why this is the case:
```
$:~/go/src/k8s.io/kubernetes$ kubectl --namespace=kube-system describe rc heapster-v10
Name:		heapster-v10
Namespace:	kube-system
Image(s):	gcr.io/google_containers/heapster:v0.18.2
Selector:	k8s-app=heapster,version=v10
Labels:		k8s-app=heapster,kubernetes.io/cluster-service=true,version=v10
Replicas:	1 current / 1 desired
Pods Status:	0 Running / 0 Waiting / 0 Succeeded / 0 Failed
No events.
```

cc @wojtek-t @fgrzadkowski @davidopp @lavalamp @dchen1107

**Comment by gmarek** [comment]
From Controller logs it looks like Kubelet was unresponsive for ~10 minutes when Heapster was using around 3GB of memory. NodeController correctly evicted all the Pods from the Node, but for some reason Heaster wasn't recreated.

**Comment by lavalamp** [comment]
How long did it take for the pod to reappear? I don't doubt that you can
catch the system like this, but it shouldn't stay like this for long.

**Comment by gmarek** [comment]
It never reappeared. I kept the cluster running for at least half an hour, but nothing happened. I have logs from controller manager if you're interested.

**Comment by lavalamp** [comment]
Can you post them here, or email to me? Thanks.

apiserver logs might also be useful.

**Comment by gmarek** [comment]
Sadly I grabbed controller-manager logs only. I'll try to reproduce it tomorrow.

**Comment by gmarek** [comment]
I wasn't able to reproduce the issue. It seems like a flake - I'll try again tomorrow, and if I'll fail I'll close this issue.

**Comment by dchen1107** [comment]
Should we close this one? cc/ @erictune this is the one I mentioned to you yesterday, but looks like it is a red herring.

**Comment by erictune** [comment]
I thought I was seeing a similar issue, but never found a smoking gun.  No objection to closing this.

**Comment by gmarek** [comment]
Yup, I failed to replicate it.
